Subject: wifi: mac80211: fix flowtable bridge acceleration for
 tx encapsulation offload capable wlan netdevices

mac80211 provides .ndo_fill_forward_path function for wlan netdevices
with "SUPPORTS_TX_ENCAP_OFFLOAD" hw capabilities, which makes software flow
offload to not use bridge acceleration, which takes a hit at
offload performance. Fill needed info for skipping bridge and return success

fix was provided by benas.bagvilas@teltonika.lt
---

--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -901,8 +901,12 @@
 	sdata = IEEE80211_DEV_TO_SUB_IF(ctx->dev);
 	local = sdata->local;
 
-	if (!local->ops->net_fill_forward_path)
-		return -EOPNOTSUPP;
+	if (!local->ops->net_fill_forward_path) {
+		path->type = DEV_PATH_ETHERNET;
+		path->dev = ctx->dev;
+		ctx->dev = NULL;
+		return 0;
+	}
 
 	rcu_read_lock();
 	switch (sdata->vif.type) {

